phoenix-core/src/main/java/org/apache/phoenix/hbase/index/Indexer.java

```diff
@@ -116,18 +110,67 @@
   private static final Log LOG = LogFactory.getLog(Indexer.class);
   private static final OperationStatus IGNORE = new OperationStatus(OperationStatusCode.SUCCESS);
   private static final OperationStatus NOWRITE = new OperationStatus(OperationStatusCode.SUCCESS);
-  
+
+  /**
+   * Class to represent pending data table rows
+   */
+  private static class PendingRow {
+      private long latestTimestamp;
+      private long count;
+      PendingRow(long latestTimestamp) {
+          count = 1;
+          this.latestTimestamp = latestTimestamp;
+      }
+
+      public void add(long timestamp) {
+          count++;
+          if (latestTimestamp < timestamp) {
+              latestTimestamp = timestamp;
+          }
+      }
+
+      public void remove() {
+          count--;
+      }
+
+      public long getCount() {
+          return count;
+      }
+
+      public long getLatestTimestamp() {
+          return latestTimestamp;
+      }
+  }
 
   protected IndexWriter writer;
+  protected IndexWriter postWriter;
+
   protected IndexBuildManager builder;
   private LockManager lockManager;
+  // The collection of pending data table rows
+  private Map<ImmutableBytesPtr, PendingRow> pendingRows = new HashMap<>();
+
+  private static boolean skipPostIndexUpdatesForTesting = false;
```

Comment:
Nit: Is there any existing naming convention for test hooks like these? That would make it easy to locate them in thr code-base. If not, we can perhaps pick a common prefix (e.g. testHook_ or testOnly_).

Reply:
I have not seen any convention on this. However, we do not use "_" on method names. Given no convention, no uses of testHook or testOnly as far as I searched, and ForTesting is very descriptive, I will keep it as it is.